grpc/handlers.go

```diff
@@ -0,0 +1,65 @@
+package grpc
+
+import (
+	"fmt"
+	"github.com/NethermindEth/juno/db"
+	"github.com/NethermindEth/juno/grpc/gen"
+	"github.com/NethermindEth/juno/utils"
+)
+
+type handlers struct {
+	db db.DB
+}
+
+func (h handlers) Tx(server gen.KV_TxServer) error {
+	dbTx := h.db.NewTransaction(false)
+	tx := newTx(dbTx)
+	defer tx.cleanup()
+
+	for {
+		cursor, err := server.Recv()
+		if err != nil {
+			return err
+		}
+
+		err = h.handleTxCursor(cursor, tx, server)
+		if err != nil {
+			return err
+		}
+	}
+}
+
+func (h handlers) handleTxCursor(
+	cur *gen.Cursor,
+	tx *tx,
+	server gen.KV_TxServer,
+) error {
+	it, err := tx.iterator(cur.Cursor)
+	if err != nil {
+		return err
+	}
+
+	switch cur.Op {
+	case gen.Op_SEEK:
+		key := utils.Flatten(cur.BucketName, cur.K)
+		it.Seek(key)
+	case gen.Op_NEXT:
+		it.Next()
```

Comment:
Just a question: how are `it.Next()` and `it.Seek(key)` being used here? Do you not need to return the results to the caller?

Reply:
For both methods return value is `bool` and I tried to follow existing proto spec (there is no field that corresponds to wrong iterator). Probably in this case it's better to return error, will check

Reply:
Was reading through the code and almost posted the same question :)

Is it better to return the error here @kirugan?

Reply:
Ok, I'll return error here. The problem with provided proto file is that there is no error => cursor correspondence, probably Rust code will add it later.

Reply:
IIRC in rust in case of error we get an empty pair response (with only the cursor id set).